wandb/util.py

```diff
@@ -1893,10 +1884,56 @@ def parse_version(version: str) -> "packaging.version.Version":
     return parse_version(version)
 
 
-def _check_wandb_core_version_compatibility(core_version: str) -> None:
-    """Checks if the installed wandb-core version is compatible with the wandb version."""
-    if parse_version(core_version) < parse_version(wandb._minimum_core_version):
-        raise ImportError(
-            f"Requires wandb-core version {wandb._minimum_core_version} or later, "
-            f"but you have {core_version}. Run `pip install --upgrade wandb-core` to upgrade."
+def get_core_path() -> str:
+    """Get the path to the wandb-core package. This is used to run the core service.
+
+    The core path can be set in the following ways:
+    The environment variable _WANDB_CORE_PATH will be used as the core path if it is set.
+    Otherwise, the wandb_core package will be imported and the core path will be retrieved from it.
+
+    Returns:
+        str: The path to the wandb-core package.
+
+    Raises:
+        Error: If the system is not supported by wandb-core, or if the installed wandb-core version is not
+                compatible with the wandb version, or if wandb-core is not installed, but required.
+    """
+    # check if the user has requesst to use wandb-core
+    if not wandb.env.is_require_core():
+        return ""
+
+    # NOTE: Environment variable _WANDB_CORE_PATH is a temporary development feature
+    #       to assist in running the core service from a live development directory.
+    core_path: str = os.environ.get("_WANDB_CORE_PATH", "")
+    # TODO: should we check version compatibility here?
+    if core_path:
+        wandb.termwarn(
+            f"Using wandb-core from path `_WANDB_CORE_PATH={core_path}`. "
+            "This is a development feature and may not work as expected."
         )
+        return core_path
+
+    required_core_version = wandb.__core_version__
+    wandb_core = get_module(
+        "wandb_core",
+        required=(
+            "`wandb-core` is required to run the core service. "
+            f"Please install it with `pip install wandb-core=={required_core_version}`."
+        ),
+    )
+
+    installed_core_version = wandb_core.__version__
+    # Check if the installed wandb-core version is compatible with the wandb version.
+    if parse_version(required_core_version) != parse_version(installed_core_version):
+        raise wandb.Error(
+            f"Requires `wandb-core` version {required_core_version}."
+            f"but you have {installed_core_version}. Please run `pip install wandb-core=={required_core_version}`."
+        )
+
+    core_path = wandb_core.get_core_path()
+    if not core_path:
+        raise wandb.Error(
+            "Looks like `wandb-core` is not compiled for your system. "
+            "Please contact support at support@wandb.com to request `wandb-core` support for your system."
+        )
```

Comment:
I wonder if users might be confused if they see this message but also see that they have `wandb-core` installed. I would also like to be cautious just in case there are weird `pip` behaviors we don't know about; what if `pip` picks the universal wheel even though a specific one is available? Yet at the same time, the wording should be accessible to data scientists.

We could have this say "Looks like wandb-core is not compiled for your system: your installed wandb-core package is using the no-op wheel. Please contact support [...]"

It would be great also to tell the user what their machine and architecture are here, so that they can tell it to support. We can include the output of [platform.platform()](https://docs.python.org/3/library/platform.html#platform.platform).

Reply:
good point about making it more accessible for data scientist, will re-write